src/Features/Core/Portable/AddImport/AddImportFixData.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Tags;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.AddImport
+{
+    internal enum AddImportFixKind
+    {
+        ProjectSymbol,
+        MetadataSymbol,
+        PackageSymbol,
+        ReferenceAssemblySymbol,
+    }
+
+    internal class AddImportFixData
+    {
+        public AddImportFixKind Kind { get; }
+
+        /// <summary>
+        /// Text changes to make to the document.  Usually just the import to add.  May also
+        /// include a change to the name node the feature was invoked on to fix the casing of it.
+        /// May be empty for fixes that don't need to add an import and only do something like
+        /// add a project/metadata reference.
+        /// </summary>
+        public ImmutableArray<TextChange> TextChanges { get; }
+
+        /// <summary>
+        /// String to display in the lightbulb menu.
+        /// </summary>
+        public string Title { get; private set; }
+
+        /// <summary>
+        /// Tags that control what glyph is displayed in the lightbulb menu.
+        /// </summary>
+        public ImmutableArray<string> Tags { get; private set; }
+
+        /// <summary>
+        /// The priority this item should have in the lightbulb list.
+        /// </summary>
+        public CodeActionPriority Priority { get; private set; }
+
+        #region When adding P2P refrences.
```

Comment:
Using derived types is currently a PITA in the serialization layer.  Have been talking to heejae about improving that.  If/when we improve things there we can move over.  

> Using a ```<remarks>``` element on each property, like this for the current one.

I'm actually very opposed to that for the reason that comments invariably become desynced from code organization.   regions are part of the code and it is immediately visible when a new field is placed into the wrong region.